### PR TITLE
Integrate examples into docs

### PR DESCRIPTION
**Pull request by nottrobin**
As discussed with @deadlight, I've done the following:

- Converted the docs in `/docs` into a Jekyll site format
- Remove superfluous stuff from `/docs` for the old documentation-builder stuff
- Removed `/en/`, and added a redirect
- Moved `/examples` into the `/docs` folder
- Switched the docs pages to point to these local examples
- Updated the root-level `package.json` so that `./run` will serve the documentation site (which now includes examples)
- Added scripts to `package.json` for `build-docs-site`, `build-docs-css` and `build-docs`
- Updated the README.md to explain how documentation now works
- Updated the `/docs/Dockerfile` to build the production image for docs.vanillaframework.io based off this new structure
- Updated `/examples/` index page to use the same styling as the rest of the docs.
- The right-hand "Contents" menu is now generated client-side with JavaScript

Fixes https://github.com/canonical-web-and-design/base-squad/issues/511

QA
--

`./run`

- Go to http://0.0.0.0:8101/, check it works.
- Browse around a bit
- Check inline examples work
- Go to http://0.0.0.0:8101/examples/, check the examples work
- Check you see the "toggle" button when viewing examples directly, but not when viewing examples on a documentation page

Now test the Docker image:

``` bash
cd docs/
docker build --build-arg COMMIT_ID=fake --tag docs .
docker run -ti -p 8901:80 docs
```

- Go to http://0.0.0.0:8901, check it works.
- Check old URLs (e.g. `http:0.0.0.0:8901/en/patterns/grid`) redirect to new ones (e.g. `http:0.0.0.0:8901/patterns/grid`)

### CODE REVIEW

**Comment by nottrobin** [comment]
Demo here: https://vanilla-framework-canonical-web-and-design-pr-2293.run.demo.haus/

The demo service spun up the demo automatically via webhook, but didn't add the comment itself. Not sure why.

**Comment by kwm14** [comment]
@nottrobin on your casualties you mention:

"The right-hand table of contents is gone. If we want it back we should probably add it with JavaScript."

It would be great to have this because I spent a lot of time rewriting and organizing the docs over the past 8 weeks, updating the titles and structure in that menu for each page 😃

**Comment by nottrobin** [comment]
@kwm14 yeah, I can see it would be nice.

It's not a tiny amount of work, and I've already taken a lot of time out of my iteration for this. Maybe I could enlist the help of @anthonydillon or @deadlight for that one? In a follow-up PR?

**Comment by kwm14** [comment]
@nottrobin ah ok not a problem at all 😉 appreciate your time and effort on this piece of work.

I will poke @deadlight or @anthonydillon to create and add back in a separate PR 👍

**Comment by nottrobin** [comment]
@kwm14 okay I went ahead and added it back. It wasn't that hard. But @anthonydillon should review it.

**Comment by deadlight** [comment]
@nottrobin thanks for the updates. I'm ready to do a full review of this now.

**Comment by nottrobin** [comment]
Thanks @deadlight. The diff is even bigger than the actual changes I made because Husky went and did things without asking =)

File: docs/_layouts/default.html [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+                <span class="u-off-screen">
+                  <a href="#">Go to the top of the page</a>
+                </span>
+              </nav>
+            </div>
+          </div>
+        </footer>
+      </main>
+
+      <aside id="side-content" class="p-aside js-contents" style="display: none">
```
Comment (deadlight) [inline docs/_layouts/default.html]: Add class "u-hide--small" to remove the aside at small screens (it currently sits under the footer)

Comment (nottrobin) [reply docs/_layouts/default.html]: Done

File: docs/examples/index.html [hunk 2]
```diff
@@ -1,22 +1,19 @@
...
 
-<div class="p-strip--image is-dark" style="background-image: url('https://assets.ubuntu.com/v1/b54d869a-vanilla-grad-background-min.png'); background-position: 75% 50%;">
-  <div class="row">
-  <h1>Vanilla framework | Examples</h1>
-</div>
-</div>
-<div class="p-strip">
-  <div class="row">
+<div class="row">
     <div class="col-3">
```
Comment (deadlight) [inline docs/examples/index.html]: Indented too far here

File: docs/settings/breakpoint-settings.md [hunk 3]
```diff
@@ -1,50 +1,49 @@
...
 
-Setting | Default value | Device example
-------------- | ------------- | -------------
-`$breakpoint-x-small`                | `460px`              | Mobile
-`$breakpoint-small`                  | `620px`              | Mobile
-`$breakpoint-medium`                 | `768px`              | Tablet
-`$breakpoint-large`                  | `1030px`             | Desktop
-`$breakpoint-navigation-threshold`   | `$breakpoint-medium` | Tablet
-`$breakpoint-heading-threshold`      | `$breakpoint-medium` | Tablet
+| Setting                            | Default value        | Device example |
```
Comment (deadlight) [inline docs/settings/breakpoint-settings.md]: :)

Comment (nottrobin) [reply docs/settings/breakpoint-settings.md]: This wasn't even me. This happened automatically.

File: docs/examples/index.html (hunk 2 above)
Comment (nottrobin) [reply docs/examples/index.html]: Fixed

**Pull request merged**